src/Spice86.Core/Emulator/Function/ExecutionFlowRecorder.cs

```diff
@@ -46,6 +45,8 @@ public class ExecutionFlowRecorder {
     private readonly HashSet<uint> _instructionsEncountered = new(200000);
     private readonly HashSet<uint> _executableCodeAreasEncountered = new(200000);
 
+    private readonly CircularBuffer<string> _callStack = new(20);
```

Comment:
This is already recorded by FunctionHandler which already provides a method to dump the call stack as string.
DumpCallStack
Isnt it what you need?

Reply:
Not quite. Comparing the output:
```....4C9B:0282 -> 1834:7251
.....1834:7259 -> 1834:6182
.....1834:7313 -> 1834:6253
......1834:626F -> 1834:623E
.......1834:624A -> 1000:13C8
........1000:13D0 -> 1000:2E10
.........1000:2E27 -> 1000:2E2D
..........1000:2E62 -> 1000:27A9
......1834:6281 -> 1000:15F2
......1834:62A6 -> 1834:623E
.......1834:624A -> 1000:13C8
........1000:13D0 -> 1000:2E10
.........1000:2E27 -> 1000:2E2D
..........1000:2E62 -> 1000:27A9
....4C9B:028D -> 1834:3EB7
.....1834:3EC4 -> 1834:39D0
.....1834:3ED8 -> 1000:06D9
......1000:070F -> 1000:3544
....4C9B:0297 -> 4C9B:047E
...2083:03D2 -> 38A2:002F
```
vs
```
- unknown_38A2_002F_38A4F expected to return to address 2083:03D7
- interrupt_handler_0x3F_36BC_04F6_370B6 expected to return to address 38AD:0027
- unknown_38AD_0025_38AF5 expected to return to address 4C9B:003F
- unknown_4C9B_0030_4C9E0 expected to return to address 4C9B:0197
- interrupt_handler_0x3F_36BC_04F6_370B6 expected to return to address 38A2:002C
- unknown_38A2_002A_38A4A expected to return to address 2083:03CB
- unknown_2083_038C_20BBC expected to return to address 2083:0E4E
- interrupt_handler_0x3F_36BC_04F6_370B6 expected to return to address 382F:0002
- interrupt_handler_0x3F_36BC_04F6_370B6 expected to return to address 396F:0022
- unknown_396F_0020_39710 expected to return to address 4B5A:057C
- unknown_382F_005C_3834C expected to return to address 2083:0E33
- unknown_2083_0E15_21645 expected to return to address 1000:015D
- unknown_1000_3298_13298 expected to return to address 1000:0260
- unknown_1000_0220_10220 expected to return to address 1000:014C
- entry_1000_0000_10000 expected to return to address
```
I did name my helper wrong. It's not the call stack, but the last 20 executes function calls.

It helps a lot in figuring out how you got to a certain location in the code. Especially to figure out where the value for an argument in the current function comes from, since it's often the return value of a previous function call.

I'm also totally fine with just keeping these in my branch if they're not of general use.

Reply:
I'm OK with merging all of this, once this _callStack object is renamed to (for example) _lastTwentyExecutedFunctionCalls ?